### ISSUE TITLE
Trailing comments aren't ignored

### ISSUE THREAD

**Issue by JRaspass** [ingydotnet/yaml-pm#143]
perl -MYAML -le 'print @{ Load "- 123 # comment\n" }'

expected: 123
got:           123 # comment

**Comment by pudge** [comment]
MOD UP

**Comment by eserte** [comment]
Comparison with other YAML parsing modules:
```
$ echo "--- something # comment" | /usr/perl5.22.0p/bin/ysh -MYAML
$VAR1 = 'something # comment';
$ echo "--- something # comment" | /usr/perl5.22.0p/bin/ysh -MYAML::XS
$VAR1 = 'something';
$ echo "--- something # comment" | /usr/perl5.22.0p/bin/ysh -MYAML::Syck
$VAR1 = 'something';
$ echo "--- something # comment" | /usr/perl5.22.0p/bin/ysh -MYAML::Tiny
$VAR1 = 'something';
```

**Comment by ambs** [comment]
Shame this isn't yet fixed :(

**Comment by 0x62ash** [comment]
Got this problem too on 1.23 version

**Comment by perlpunk** [comment]
In February I implemented trailing comments in my fork.
See https://github.com/perlpunk/yaml-pm/tree/trailing-comments

However, they would need to be optional, otherwise it would break people's code if they rely on the current behaviour.
So, the code would need to changed using an option like `$YAML::TrailingComments`, and the tests also need to be changed back, and new tests need to be added.

If anyone wants to do it, feel free to use my branch as a base.
I don't want to put too much work into YAML.pm, because I hope https://metacpan.org/pod/YAML::PP will be a good replacement soon.

**Comment by ambs** [comment]
Given the spec says comments are valid, I think the default behavior would be to parse the trailing comments as that...

**Comment by perlpunk** [comment]
@ambs I don't want to break any code relying on that. same for @ingydotnet.
The only thing I can imagine is introducing an option that will be false for now, and after a certain amount of time, set to true.

**Comment by ambs** [comment]
Suggestion: `use YAML 'strict';`

**Comment by perlpunk** [comment]
I'm at the Perl Toolchain summit and have been working on this.

* https://github.com/ingydotnet/yaml-pm/pull/189
* https://github.com/ingydotnet/yaml-pm/pull/190

**Comment by perlpunk** [comment]
Released YAML 1.24_001
